include/morphio/iterators.h

```diff
@@ -27,4 +30,25 @@ class Iterator
     Iterator& operator++();
     Iterator operator++(int);
 };
-} // namespace morphio
+
+class graph_iterator
+{
+    friend class vasculature::Section;
+    friend class vasculature::Vasculature;
+    std::set<vasculature::Section> visited;
+
+    std::stack<vasculature::Section> container;
```

Comment:
I think I would prefer a queue rather than a stack to get a breadth first pattern